clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -7085,18 +7102,24 @@ ExprResult Sema::CheckTemplateArgument(NonTypeTemplateParmDecl *Param,
                                    ? Context.getIntWidth(IntegerType)
                                    : Context.getTypeSize(IntegerType));
 
-      SugaredConverted = TemplateArgument(Context, Value, ParamType);
-      CanonicalConverted =
-          TemplateArgument(Context, Value, Context.getCanonicalType(ParamType));
-      return ArgResult;
+      if (ArgPE) {
```

Comment:
So we had a static analysis tool claim that if we get there `ArgPE` will always be `nullptr` based on just manual inspection it was not clear to me this was the case. 

So I decided to change the code based on this assumption and see if anything breaks and nothing broke.

At minimum we are not covering this code in testing which is not ideal. Can you come up with a test case that hits this or maybe show that this indeed dead code?

Reply:
Right, that's not possible. At some point in the development of this patch, I changed the `StrictCheck` mode to always use the C++17 rules for converted constant expressions, and so the changes to this function from this point forward became unnecessary, but I must have forgotten to go back and remove this.

I posted a patch to clean this up here: https://github.com/llvm/llvm-project/pull/158515